utils/swift_build_support/swift_build_support/products/product.py

```diff
@@ -237,42 +237,48 @@ def is_cross_compile_target(self, host_target):
         return self.args.cross_compile_hosts and \
             host_target in self.args.cross_compile_hosts
 
-    def generate_darwin_toolchain_file(self, platform, arch):
-        shell.makedirs(self.build_dir)
-        toolchain_file = os.path.join(self.build_dir, 'BuildScriptToolchain.cmake')
-
-        cmake_osx_sysroot = xcrun.sdk_path(platform)
-
+    def target_for_platform(self, platform, arch):
         target = None
         if platform == 'macosx':
             target = '{}-apple-macosx{}'.format(
                 arch, self.args.darwin_deployment_version_osx)
         elif platform == 'iphonesimulator':
-            target = '{}-apple-ios{}'.format(
+            target = '{}-apple-ios{}-simulator'.format(
                 arch, self.args.darwin_deployment_version_ios)
         elif platform == 'iphoneos':
             target = '{}-apple-ios{}'.format(
                 arch, self.args.darwin_deployment_version_ios)
         elif platform == 'appletvsimulator':
-            target = '{}-apple-tvos{}'.format(
+            target = '{}-apple-tvos{}-simulator'.format(
                 arch, self.args.darwin_deployment_version_tvos)
         elif platform == 'appletvos':
             target = '{}-apple-tvos{}'.format(
                 arch, self.args.darwin_deployment_version_tvos)
         elif platform == 'watchsimulator':
-            target = '{}-apple-watchos{}'.format(
+            target = '{}-apple-watchos{}-simulator'.format(
                 arch, self.args.darwin_deployment_version_watchos)
         elif platform == 'watchos':
             target = '{}-apple-watchos{}'.format(
                 arch, self.args.darwin_deployment_version_watchos)
-        else:
-            raise RuntimeError("Unhandled platform?!")
+        return target
+
+    def generate_darwin_toolchain_file(self, platform, arch):
+        shell.makedirs(self.build_dir)
+        toolchain_file = os.path.join(self.build_dir, 'BuildScriptToolchain.cmake')
+
+        cmake_osx_sysroot = xcrun.sdk_path(platform)
+
+        target = self.target_for_platform(platform, arch)
+        if not target:
+            raise RuntimeError('Unhandled platform {}?!'.format(platform))
 
         toolchain_args = {}
 
         toolchain_args['CMAKE_SYSTEM_NAME'] = 'Darwin'
         toolchain_args['CMAKE_OSX_SYSROOT'] = cmake_osx_sysroot
         toolchain_args['CMAKE_OSX_ARCHITECTURES'] = arch
+        toolchain_args['CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG'] = '\"-Wl,-rpath,\"'
```

Comment:
Is this changing current behavior? What rpath do we end up with on Darwin here? Can you make sure that the artifacts are the same before/after?